packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -762,7 +762,12 @@ describe('TransactionController Integration', () => {
               '0x1',
             ),
             buildEthBlockNumberRequestMock('0x3'),
-            buildEthGetTransactionReceiptRequestMock('0x1', '0x1', '0x3'),
+            buildEthGetTransactionReceiptRequestMock(
+              '0x1',
+              '0x1',
+              '0x3',
+              '0x2',
```

Comment:
The sporadic failure was caused by the duration of the test sometimes allowing the `PendingTransactionTracker` to mark the first transaction as `confirmed`.

This meant that when processing the second transaction, the nonce tracker didn't find any pending transactions, meaning it looked at the network for the highest nonce which was mocked to always return `0x1`. This in turn meant the `data` included the wrong nonce and so wasn't recognised in the mocks.

The simplest fix was changing the receipt mocks to use an unrecognised status meaning the first mock would always stay as `submitted` and ensure the second nonce is incremented correctly.

Tested 1000 times via a script, no failures. Tested before fix also and was hitting it 14 / 100 times.